dashboard/src/components/Catalog/Catalog.tsx

```diff
@@ -94,7 +95,10 @@ export default function Catalog() {
   const dispatch = useDispatch();
 
   const [filters, setFilters] = React.useState(initialFilterState());
-  const [page, setPage] = React.useState(0);
+  // pageNum is only used to avoid flicker in the CatalogItems.
+  // It is no longer used for pagination which is handled by the server's
+  // opaque nextPageToken etc.
+  const [pageNum, setPageNum] = React.useState(0);
```

Comment:
We might want to consider a map of triggered requests and their status? This way, when unmounting the component, we could cancel any unnecessary ongoing requests. Besides, it could help us avoid duplicated requests being fired at the same time?